axelor-human-resource/src/main/java/com/axelor/apps/hr/web/EmployeeController.java

```diff
@@ -62,24 +70,43 @@ public void showAnnualReport(ActionRequest request, ActionResponse response)
     String name =
         I18n.get("Annual expenses report") + " :  " + user.getFullName() + " (" + yearName + ")";
 
+    Employee employee = Beans.get(EmployeeRepository.class).find(Long.valueOf(employeeId));
+    BirtTemplate annualReportBirtTemplate = getAnnualReportBirtTemplate(employee);
     String fileLink =
-        ReportFactory.createReport(IReport.EMPLOYEE_ANNUAL_REPORT, name)
-            .addParam("EmployeeId", Long.valueOf(employeeId))
-            .addParam(
-                "Timezone",
-                getTimezone(
-                    Beans.get(EmployeeRepository.class).find(Long.valueOf(employeeId)).getUser()))
-            .addParam("YearId", Long.valueOf(yearId))
-            .addParam("Locale", ReportSettings.getPrintingLocale(null))
-            .toAttach(Beans.get(EmployeeRepository.class).find(Long.valueOf(employeeId)))
-            .generate()
-            .getFileLink();
+        Beans.get(BirtTemplateService.class)
+            .generateBirtTemplateLink(
+                annualReportBirtTemplate,
+                EntityHelper.getEntity(employee),
+                Map.of("YearId", Long.valueOf(yearId)),
+                name,
+                annualReportBirtTemplate.getAttach(),
+                annualReportBirtTemplate.getFormat());
 
     response.setView(ActionView.define(name).add("html", fileLink).map());
 
     response.setCanClose(true);
   }
 
+  private BirtTemplate getAnnualReportBirtTemplate(Employee employee) throws AxelorException {
+    BirtTemplate employeeAnnualReportBirtTemplate =
+        Optional.ofNullable(employee.getUser())
+            .map(User::getActiveCompany)
+            .filter(Objects::nonNull)
+            .map(Company::getHrConfig)
+            .filter(Objects::nonNull)
+            .map(HRConfig::getEmployeeAnnualReportBirtTemplate)
+            .filter(Objects::nonNull)
+            .orElse(null);
+
+    if (ObjectUtils.isEmpty(employeeAnnualReportBirtTemplate)
+        || ObjectUtils.isEmpty(employeeAnnualReportBirtTemplate.getTemplateMetaFile())) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(BaseExceptionMessage.BIRT_TEMPLATE_CONFIG_NOT_FOUND));
+    }
+    return employeeAnnualReportBirtTemplate;
+  }
```

Comment:
Please move this method to a service, also please call `HrConfigService` to get the configuration.